src/main/scala/jawnfs2/package.scala

```diff
@@ -60,27 +61,36 @@ package object jawnfs2 {
       * @tparam J the JSON AST to return
       * @param mode the async mode of the Jawn parser
       */
-    def parseJson[J](mode: AsyncParser.Mode)(implicit absorbable: Absorbable[O], facade: Facade[J]): Stream[F, J] =
+    def parseJson[J](mode: AsyncParser.Mode)(implicit absorbable: Absorbable[O], facade: RawFacade[J]): Stream[F, J] =
       stream.through(jawnfs2.parseJson(mode))
 
     /**
-      * Parses the source to a single JSON value.  If the stream is empty, parses to
-      * the facade's concept of jnull.
+      * Parses the source to a single JSON optional JSON value.
+      *
+      * @param facade the Jawn facade to materialize `J`
+      * @tparam J the JSON AST to return
+      * @return some parsed JSON value, or None if the source is empty
+      */
+    def runJsonOption[J](implicit F: Sync[F], absorbable: Absorbable[O], facade: RawFacade[J]): F[Option[J]] =
+      stream.parseJson(AsyncParser.SingleValue).compile.last
+
+    /**
+      * Parses the source to a single JSON value.
       *
       * @param facade the Jawn facade to materialize `J`
       * @tparam J the JSON AST to return
       * @return the parsed JSON value, or the facade's concept of jnull if the source is empty
       */
     def runJson[J](implicit F: Sync[F], absorbable: Absorbable[O], facade: Facade[J]): F[J] =
-      stream.parseJson(AsyncParser.SingleValue).compile.fold(facade.jnull())((_, json) => json)
+      runJsonOption.map(_.getOrElse(facade.jnull()))
```

Comment:
Maybe we should deprecate this one.  Was `null` ever really a sensible default / empty / zero JSON value in the first place?

Reply:
I am uncertain. technically using an optional decoder of a top level object the two valid states are nothing and `null`, as a result I feel inclined this may still have a place.

Reply:
I'm not real fond of a default value like this where there isn't a monoid.  Why is `null` any more of a default than `""` or `false` or `{}` or `[]` or `0`?